tests/mmm/test_budget_optimizer.py

```diff
@@ -135,72 +120,97 @@ def test_allocate_budget(
                 },
                 "channels": ["channel_1", "channel_2"],
             },
+            None,
             {"channel_1": 0.0, "channel_2": 7.94e-13},
             2.38e-10,
         ),
     ],
+    ids=["default_minimizer_kwargs", "custom_minimizer_kwargs", "zero_total_budget"],
 )
-def test_allocate_budget_zero_total(
-    total_budget, budget_bounds, parameters, expected_optimal, expected_response
+def test_allocate_budget(
+    total_budget,
+    budget_bounds,
+    parameters,
+    minimize_kwargs,
+    expected_optimal,
+    expected_response,
+    dummy_df,
 ):
-    adstock = GeometricAdstock(l_max=4)
-    saturation = LogisticSaturation()
+    df_kwargs, X_dummy, y_dummy = dummy_df
+
+    mmm = MMM(
+        adstock=GeometricAdstock(l_max=4),
+        saturation=LogisticSaturation(),
+        **df_kwargs,
+    )
+
+    mmm.build_model(X=X_dummy, y=y_dummy)
+
+    # Only these parameters are needed for the optimizer
+    mmm.idata = az.from_dict(
+        posterior={
+            "saturation_lam": parameters["saturation_params"]["lam"],
+            "saturation_beta": parameters["saturation_params"]["beta"],
+            "adstock_alpha": parameters["adstock_params"]["alpha"],
+        }
+    )
 
+    # Create BudgetOptimizer Instance
     optimizer = BudgetOptimizer(
-        adstock=adstock,
-        saturation=saturation,
+        model=mmm,
         num_periods=30,
-        parameters=parameters,
-        adstock_first=True,
-        scales=np.array([1, 1]),
     )
-    match = "Using default equality constraint"
-    with pytest.warns(UserWarning, match=match):
-        optimal_budgets, total_response = optimizer.allocate_budget(
-            total_budget, budget_bounds
-        )
-    assert optimal_budgets == pytest.approx(expected_optimal, rel=1e-2)
-    assert -total_response.fun == pytest.approx(expected_response, abs=1e-1)
+
+    # Allocate Budget
+    # match = "Using default equality constraint"
+    # with pytest.warns(UserWarning, match=match):
```

Comment:
do you want to remove these commented lines?

Reply:
No, we can keep them